kernel/src/scan/mod.rs

```diff
@@ -320,7 +320,24 @@ pub enum ColumnType {
     Partition(usize),
 }
 
-pub type ScanData = (Box<dyn EngineData>, Vec<bool>);
+/// A transform is ultimately a `Struct` expr. This holds the set of expressions that make that struct expr up
+type Transform = Vec<TransformExpr>;
+
+/// Transforms aren't computed all at once. So static ones can just go straight to `Expression`, but
+/// things like partition columns need to filled in. This enum holds an expression that's part of a
+/// `Transform`.
+pub(crate) enum TransformExpr {
+    Static(Expression),
+    Partition(usize),
+}
+
+// TODO(nick): Make this a struct in a follow-on PR
+// (data, deletion_vec, transforms)
+pub type ScanData = (
+    Box<dyn EngineData>,
+    Vec<bool>,
+    HashMap<usize, ExpressionRef>,
```

Comment:
Just like CDF, another case where we want to return maps to the engine. Would be curious to hear what the plan for engine integration/FFI is.